### PR TITLE
[MPOM-354] leverage new report plugin

### PR DESCRIPTION
**Pull request by kwin**
use javadoc from plugin documentation

### CODE REVIEW

File: maven-plugins/pom.xml [hunk 1]
```diff
@@ -145,9 +148,21 @@ under the License.
...
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <reportSets>
+          <reportSet>
+            <id>default</id>
+            <reports>
+              <report>javadoc</report>
+            </reports>
+          </reportSet>
+        </reportSets>
+      </plugin>
```
Comment (slawekjaranowski) [inline maven-plugins/pom.xml:162]: we already have such report in maven-parent, 
it is in reporting profile ... so maybe also maven-plugin-report-plugin should be in reporting profile

Comment (kwin) [reply maven-plugins/pom.xml:162]: why the separate profile? What is the site lifecycle supposed to generate without that profile?

Comment (slawekjaranowski) [reply maven-plugins/pom.xml:162]: Separate profile only to be consistent how we do it now.

It is, what I mention from time to time - I think that we have to many reports, and we introduce one simple site build and one with full reports with profile - but consensus is difficult to achieve.
Sometime during release special profile can be forgotten....

Comment (kwin) [reply maven-plugins/pom.xml:162]: Ok, so what about moving all reports (incl. `maven-plugin-report-plugin` and `maven-project-info-reports-plugin`) to the profile `reporting` and always activate that profile during (in addition to `apache-release`) in `release:perform` (https://maven.apache.org/maven-release/maven-release-plugin/perform-mojo.html#releaseProfiles)? Then only the manually maintained pages are built during a regular `site` execution.

Comment (kwin) [reply maven-plugins/pom.xml]: Done in https://github.com/apache/maven-parent/pull/90/commits/82990e3a4c6c0f9896a0a32dd66d58ef0ee7b670

File: pom.xml [hunk 2]
```diff
@@ -1080,6 +1080,7 @@ under the License.
           <artifactId>maven-release-plugin</artifactId>
           <configuration>
             <autoVersionSubmodules>true</autoVersionSubmodules>
+            <releaseProfiles>apache-release,reporting</releaseProfiles>
```
Comment (slawekjaranowski) [inline pom.xml:1083]: Should be for next PR - it is out of scope for me

File: pom.xml [hunk 3]
```diff
@@ -1212,21 +1213,8 @@ under the License.
...
-              <report>scm</report>
-              <report>issue-management</report>
-              <report>mailing-lists</report>
-              <report>dependency-management</report>
-              <report>dependencies</report>
-              <report>dependency-convergence</report>
-              <report>ci-management</report>
-              <report>plugin-management</report>
-              <report>plugins</report>
-              <report>distribution-management</report>
```
Comment (slawekjaranowski) [inline pom.xml:1229]: Should be for next PR - it is out of scope for me

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (kwin) [reply pom.xml]: Created https://issues.apache.org/jira/browse/MPOM-358 and removed from this PR.

File: pom.xml (hunk 3 above)
Comment (kwin) [reply pom.xml]: Created https://issues.apache.org/jira/browse/MPOM-358 and removed from this PR.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**